CMakeLists.txt

```diff
@@ -53,18 +53,19 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(SONATA_REPORT_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/include)
 set(SONATA_REPORT_SRC_DIR ${PROJECT_SOURCE_DIR}/src)
 
-if(SONATA_REPORT_ENABLE_WARNING_AS_ERROR)
-  set(SONATA_REPORT_COMPILE_OPTIONS -Wall -Wextra -pedantic -Werror)
-else()
-  set(SONATA_REPORT_COMPILE_OPTIONS -Wall -Wextra -pedantic)
-endif()
-
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
     set(ENABLE_COVERAGE ON)
+    set(SONATA_REPORT_ENABLE_WARNING_AS_ERROR ON)
 else()
     set(ENABLE_COVERAGE OFF)
 endif()
 
+if(SONATA_REPORT_ENABLE_WARNING_AS_ERROR)
+  set(SONATA_REPORT_COMPILE_OPTIONS -Wall -Wextra -pedantic -Werror)
+else()
+  set(SONATA_REPORT_COMPILE_OPTIONS -Wall -Wextra -pedantic)
+endif()
```

Comment:
We should check compiler id here before adding flags? (as reference, see [here](https://github.com/neuronsimulator/nrn/blob/ee63557e2b3728c0a61547bb18c929b596b2d516/cmake/CompilerFlagsHelpers.cmake#L14))

Reply:
By the way, nrn cmake was derived from [highfive](https://github.com/BlueBrain/HighFive/blob/master/CMake/config/CompilerFlagsHelpers.cmake). I don't think we need detailed one like that but there should be minimum check before inserting a compiler flag.